File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Models/UDS/UDSRepositoryModel.ts
import UDSRepositoryStatus = require('App/Models/UDS/UDSRepositoryStatus');
import ContainerModel = require('App/Models/Commons/ContainerModel');

class UDSRepositoryModel {
    constructor() {       
    }

    UniqueId: string;
    Name: string;
    SequenceCurrentYear: number;
    SequenceCurrentNumber: number;
    ModuleXML: string;
    Version: number;
    RegistrationDate: Date;
    ActiveDate: Date;
    ExpiredDate: Date;
    DSWEnvironment: number;
    Alias: string;
    Status: UDSRepositoryStatus;
    Container: ContainerModel;
}

export = UDSRepositoryModel;

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Model/Entities/UDS/UDSRepositoryModel.cs
using System;

namespace VecompSoftware.DocSuiteWeb.Model.Entities.UDS
{
    public class UDSRepositoryModel
    {
        #region [ Constructor ]
        public UDSRepositoryModel()
        {

        }

        public UDSRepositoryModel(Guid idRespository)
        {
            Id = idRespository;
        }
        #endregion

        #region [ Properties ]

        public DateTimeOffset? ActiveDate { get; set; }

        public DateTimeOffset? ExpiredDate { get; set; }

        public short SequenceCurrentYear { get; set; }

        public int SequenceCurrentNumber { get; set; }

        public Guid Id { get; set; }

        public string ModuleXML { get; set; }

        public string Name { get; set; }

        public UDSRepositoryStatus? Status { get; set; }

        public UDSSchemaRepositoryModel SchemaRepository { get; set; }

        public short? Version { get; set; }

        public int DSWEnvironment { get; set; }

        public string Alias { get; set; }

        #endregion

        #region [ Methods ]

        public bool HasId()
        {
            return !Id.Equals(Guid.Empty);
        }

        #endregion
    }
}
